scripts.js

```diff
@@ -266,19 +258,23 @@ function handleDropdownRegion() {
     fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
   }
   
-  // Get currently selected region if in the same region page based on localeHome value
-  const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
-  if (!currentRegion.length) {
-    // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
-    sessionStorage.getItem('blog-selected-language') || window.blog.language;
-  } else {
-      // uk and apac, just save the info in the session storage
-      sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
-  }
+  // get actual selected Region
+  getActualRegion = function() {
+    const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
+    let selectedLocale;
+    if (!currentRegion.length) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
+      selectedLocale = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+    } else {
+      selectedLocale = currentRegion[0].lang;
+        // uk and apac, just save the info in the session storage
+        sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
+    } return { selectedLocale };
+  } ();
 
   // Automatically build the dropdown based on Locale List
   const dropdownRegionList = document.querySelector('.region-dropdown-list');
-  const selectedRegionLocale = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+  const selectedRegionLocale =  getActualRegion.selectedLocale;
   if (dropdownRegionList) {
     for (const {lang: locale, localeName: localeName, localeHome: localeHome} of regionsNameList) {
```

Comment:
If the property name is the same as the variable name you want to use later, you can skip the assignment:
```suggestion
    for (const { lang: locale, localeName, localeHome } of regionsNameList) {
```